app/models/solidus_stripe/gateway.rb

```diff
@@ -51,28 +58,13 @@ def authorize(amount_in_cents, source, options = {})
     #
     # @todo add support for capturing custom amounts
     def capture(amount_in_cents, payment_intent_id, options = {})
-      payment = options[:originator] or
-        raise ArgumentError, "please provide a payment with the :originator option"
-
-      unless payment_intent_id
-        raise ArgumentError, "missing payment_intent_id"
-      end
-
-      unless amount_in_cents == payment.display_amount.cents
-        raise \
-          "Capturing a custom amount is not supported yet, " \
-          "tried #{amount_in_cents} but can only accept #{payment.display_amount.cents}."
-      end
-
-      unless payment_intent_id.start_with?('pi_')
-        raise ArgumentError, "the payment-intent id has the wrong format"
-      end
-
-      payment_intent = request { Stripe::PaymentIntent.capture(payment_intent_id) }
+      check_given_amount_matches_order_total(amount_in_cents, options)
+      check_payment_intent_id(payment_intent_id)
 
+      payment_intent = capture_stripe_payment_intent(payment_intent_id)
       build_payment_log(
         success: true,
-        message: "Capture was successful",
+        message: "PaymentIntent was confirmed successfully",
         response_code: payment_intent.id,
         data: payment_intent,
       )
```

Comment:
Not strictly related, but I was thinking about what happens in case of exceptions, and I noticed that we're logging a successful log with `build_payment_log(success: true, ...)`, but there might be cases where `capture_stripe_payment_intent` raises an error, and we handle only `Stripe::InvalidRequestError`.

Are we sure that this is the only exception to handle? I was looking for other possible exceptions, and maybe we can handle a more generic [`Stripe::StripeError`](https://github.com/stripe/stripe-ruby/blob/980b7b011f6fbff276ecadce5dd884e00669724f/lib/stripe/errors.rb#L6) at all the points when we're doing these external calls 🤔 WDYT?

If you believe this concern is not directly related to the changes proposed in this PR, we can discuss it at a later time 👍

Reply:
@rainerdema, yeah, it looks like, for now, only the happy path is covered in some situations, and I'm also unsure whether `InvalidRequestError` handles all possible failures. That's definitely food for another PR (probably related to #224)